app/models/explainer.rb

```diff
@@ -108,23 +103,30 @@ def self.update_paragraphs_in_alegre(id, previous_paragraphs_count, timestamp)
   end
 
   def self.search_by_similarity(text, language, team_id)
+    models_thresholds = Explainer.get_alegre_models_and_thresholds(team_id)
     context = {
       type: 'explainer',
       team: Team.find(team_id).slug
     }
     context[:language] = language unless language.nil?
     params = {
       text: text,
-      models: ALEGRE_MODELS_AND_THRESHOLDS.keys,
-      per_model_threshold: ALEGRE_MODELS_AND_THRESHOLDS,
+      models: models_thresholds.keys,
+      per_model_threshold: models_thresholds,
       context: context
+
     }
     response = Bot::Alegre.query_sync_with_params(params, "text")
     results = response['result'].to_a.sort_by{ |result| result['_score'] }
     explainer_ids = results.collect{ |result| result.dig('context', 'explainer_id').to_i }.uniq.first(3)
     explainer_ids.empty? ? Explainer.none : Explainer.where(team_id: team_id, id: explainer_ids)
   end
 
+  def self.get_alegre_models_and_thresholds(team_id)
+    tbi = Bot::Alegre.get_alegre_tbi(team_id)
+    tbi&.get_alegre_models_and_thresholds || { Bot::Alegre::PARAPHRASE_MULTILINGUAL_MODEL => 0.7 }
```

Comment:
Did you confirm that `get_alegre_models_and_thresholds` return that format (`model => threshold`)?

Reply:
I expected that we store this format in settings can you please let me know how can I confirm this one or send me a workspace(QA or Live) that has this values stored on DB settings.

Reply:
Yes, please confirm. You can do that even in your local environment. Just update similarity settings using the UI (e.g., `/:workspace-slug/settings/similarity`) and then check it in Rails console.

Reply:
@DGaffney please advice if there is there a method in Check API Alegre Bot that reads the settings and returns a hash with the models and thresholds to be used for similarity matching

Reply:
```
irb(main):009:0> tbi.settings
=> 
{"master_similarity_enabled"=>true,
 "text_similarity_enabled"=>true,
 "image_similarity_enabled"=>true,
 "video_similarity_enabled"=>true,
 "audio_similarity_enabled"=>false,
 "date_similarity_threshold_enabled"=>false,
 "transcription_similarity_enabled"=>true,
 "single_language_fact_checks_enabled"=>false,
 "text_length_matching_threshold"=>6,
 "text_elasticsearch_matching_threshold"=>0.93,
 "text_elasticsearch_suggestion_threshold"=>0.85,
 "text_vector_matching_threshold"=>0.95,
 "text_vector_suggestion_threshold"=>0.8,
 "image_hash_matching_threshold"=>0.86,
 "image_hash_suggestion_threshold"=>0.7,
 "video_hash_matching_threshold"=>0.95,
 "video_hash_suggestion_threshold"=>0.75,
 "audio_hash_matching_threshold"=>0.95,
 "audio_hash_suggestion_threshold"=>0.9,
 "transcription_minimum_duration"=>20,
 "transcription_maximum_duration"=>300,
 "transcription_minimum_requests"=>0,
 "text_similarity_model"=>["elasticsearch", "xlm-r-bert-base-nli-stsb-mean-tokens", "paraphrase-multilingual-mpnet-base-v2"],
 "alegre_model_in_use"=>["elasticsearch", "xlm-r-bert-base-nli-stsb-mean-tokens", "paraphrase-multilingual-mpnet-base-v2"],
 "similarity_date_threshold"=>"1",
 "language_for_similarity"=>"auto",
 "min_es_score"=>0}

```

These are available directly from `TeamBotInstallation.find(x).settings`- we have a more scoped `TeamBotInstallation.find(x).alegre_settings` that should contain anything relevant for this context!

Reply:
@DGaffney is there any method in Alegre Bot that parses these settings and returns the model and threshold to use for text similarity matching?

Reply:
@DGaffney pushed a commit to handle callback for alegre settings